clients/include/hipblas_test.hpp

```diff
@@ -0,0 +1,172 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IM-
+ * PLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNE-
+ * CTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * ************************************************************************ */
+
+#pragma once
+
+#ifdef GOOGLE_TEST
+#include <gtest/gtest.h>
+#endif
+
+#include "hipblas_arguments.hpp"
+
+#include <cstdio>
+#include <cstring>
+#include <functional>
+#include <future>
+#include <iostream>
+#include <mutex>
+#include <queue>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#ifdef WIN32
+typedef long long ssize_t; /* x64 only supported */
+#endif
+
+// ----------------------------------------------------------------------------
+// Normal tests which return true when converted to bool
+// ----------------------------------------------------------------------------
+struct hipblas_test_valid
+{
+    // Return true to indicate the type combination is valid, for filtering
+    virtual explicit operator bool() final
+    {
+        return true;
+    }
+
+    // Require derived class to define functor which takes (const Arguments &)
+    virtual void operator()(const Arguments&) = 0;
+
+    virtual ~hipblas_test_valid() = default;
+};
+
+// ----------------------------------------------------------------------------
+// Error case which returns false when converted to bool. A void specialization
+// of the FILTER class template above, should be derived from this class, in
+// order to indicate that the type combination is invalid.
+// ----------------------------------------------------------------------------
+struct hipblas_test_invalid
+{
+    // Return false to indicate the type combination is invalid, for filtering
+    virtual explicit operator bool() final
+    {
+        return false;
+    }
+
+    // If this specialization is actually called, print fatal error message
+    virtual void operator()(const Arguments&) final
+    {
+        static constexpr char msg[] = "Internal error: Test called with invalid types";
+
+#ifdef GOOGLE_TEST
+        FAIL() << msg;
+#else
+        std::cerr << msg << std::endl;
+        abort();
+#endif
+    }
+
+    virtual ~hipblas_test_invalid() = default;
+};
+
+/* ============================================================================================ */
+/*! \brief  Normalized test name to conform to Google Tests */
+// The template parameter is only used to generate multiple instantiations with distinct static local variables
+template <typename>
+class RocBLAS_TestName
```

Comment:
A couple instances of rocblas remaining instead of hipblas.

Reply:
Thanks that may be dead code, will fix.  Sorry for the PR size it just kept growing 🤣